src/flamegpu/util/JitifyCache.cu

```diff
@@ -218,13 +220,28 @@ std::unique_ptr<KernelInstantiation> JitifyCache::loadKernel(const std::string &
     const std::string arch = std::to_string((status == cudaSuccess) ? util::compute_capability::getComputeCapability(currentDeviceIdx) : 0);
     status = cudaRuntimeGetVersion(&currentDeviceIdx);
     const std::string cuda_version = std::to_string((status == cudaSuccess) ? currentDeviceIdx : 0);
+    const std::string seatbelts = std::to_string(SEATBELTS);
+    std::string debug = std::string();
+#ifdef __CUDACC_DEBUG__
+    debug += std::to_string(__CUDACC_DEBUG__);
```

Comment:
What do these macros actually evaluate to?

If they all evaluate to `1` when set otherwise not written. You end up with the issue where different combinations could lead to the same  short referenced.

I'm a bit reluctant with regards to adding all this debug trash to short reference (e.g. if it inflates the file name/path too long for certain OS/s). If they're really necessary add them to the long reference. This will mean if the long reference doesn't match the cash state is overwritten. But there's no reason that DEBUG/NDEBUG should differ.

Reply:
tldr: different short reference means different cache file, different long reference means cache file is overwritten.

Reply:
They probably don't actually have a value thinking about it. 
The debug/release state is important though. If your release build uses a cache that was build with debug you're not going to be happy about runtime. 

I just chucked them in short ref as suggested for the SEATBELTS value. I'm not familiar enough with the caching to know the difference.
Will sort it tomorrow.

Reply:
It would be a bit more work, but tweaking it so that the entire options vector is added to the long cache would make most sense.

Reply:
This way, changes to options in future will always be considered.

Reply:
Yeah I only specified short_ref for seatbelts because we switch between those two options that having seperate cache files would be useful. The debug flags are internal stuff which shouldn't change ever/often within a specific cache file. So they're better off in long reference as there would be not reason to differentiate cache files.